### PR TITLE
Documentation MVP

### PR DESCRIPTION
**Pull request by chloerice**
## TL;DR: 
This PR adds bare minimum documentation to the repo for the tokens and components packages and the project itself. 

## Approaches
There's three approaches we can take. I personally vote for the second approach, since the READMEs will auto show in the detail pages for the NPM packages when shipped and the content is co-located with the code it is documenting. Click the links below to experience each approach, then cast your vote 📬 

|  Approach | Pros  | Cons |
|---|---|---|
|[Approach 1 - Single README at the base of the project holds everything](https://github.com/Shopify/polaris/blob/e9b896f6870aaf67aa405f0e4f18597afca2d7b2/README.md)|No need to navigate to other pages to read the docs you need|Overwhelming wall of text; Have to do a lot of scrolling up and down or Cmd + F to re-find the content you're looking for after initially clicking a link in the table of contents at the top|
|[Approach 2 - Single README at the base of the project, READMEs at the base of each of the packages](https://github.com/Shopify/polaris/blob/d4a71041250360588ba4dc278ed3012fb1134cea/README.md)|Package docs close to their source code; Separation of concerns; Cross-links where relevant|Makes contribution to the docs less straight forward (not really relevant since the README approach is a temporary docs solution)|
|[Approach 3 - Single README at the base of the project, additional Markdown files in a `documentation` folder at the base of the project](https://github.com/Shopify/polaris/blob/9ba0d2e9c47dc091a59a9a7bda084f88a3fae95e/README.md)|Clear place to look for and contribute to docs|Docs not co-located with their source code|

### Component props
I've opted to not directly document the props of the components because they are experimental and subject to change. I'm going to add a link to the props definitions in the source code to the components table instead so there's no need to update in multiple places as things change.

### CODE REVIEW

File: documentation/components.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+| **borderStyle** | “auto”, “none”, “dotted”, “dashed”, “solid”, “double”, “groove”, “ridge”, “inset”, “outset” |
+
+#### Layout
+
+| Prop name                       | Values accepted                                                                                          |
+| ------------------------------- | -------------------------------------------------------------------------------------------------------- |
+| **position**                    | “absolute”, “relative”, “static”, “sticky”, “initial”, “fixed”                                           |
+| **display**                     | “block”, “inline”, “inline-block”, “flex”, “inline-flex”, “grid”, “inline-grid”, “flow-root”, “contents” |
+| **top / left / right / bottom** | _Accepts any [size or space atomic value](#space-and-size-values) value._                                |
+| **overflow**                    | “visible”, “hidde”, “clip”, “scroll”, “auto”, “initial”                                                  |
```
Comment (chloerice) [inline documentation/components.md:100]:
```suggestion
| **overflow**                    | “visible”, “hidden”, “clip”, “scroll”, “auto”, “initial”                                                  |
```

File: documentation/components.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
+# @polaris/components
+
+[Design tokens](tokens.md) | [Atoms](#atoms-shortcuts-helpers) | [Components](#component-api-reference) |
+
+## Atoms (shortcuts, helpers)
+
+[Source code](https://github.com/Shopify/polaris/blob/master/packages/components/src/atoms/sprinkles.css.ts)
+
+Atoms are the smallest functional elements of the design foundation system. Using Sprinkles, we apply the [design tokens](packages/tokens/README.md) to common CSS properties alongside their natively accepted values and convert them into atoms. These atoms encapsulate common CSS rules into global classes to reduce redundant styles. Atoms are available in your React code in two ways: through props on each of our components and through the atoms function for use in .css.ts files to create custom classes.
```
Comment (chloerice) [inline documentation/components.md:9]:
```suggestion
Atoms are the smallest functional elements of the design foundation system. Using [Sprinkles](https://vanilla-extract.style/documentation/sprinkles-api/), we apply the [design tokens](packages/tokens/README.md) to common CSS properties alongside their natively accepted values and convert them into atoms. These atoms encapsulate common CSS rules into global classes to reduce redundant styles. Atoms are available in your React code in two ways: through props on each of our components and through the atoms function for use in `.css.ts` files to create custom classes.
```

File: documentation/components.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+When an Prop name is set on a component, the HTML renders with the corresponding atomic classes. For example, the marginTop prop accepts any of the spacing tokens. Setting the marginTop prop to 4 on a Box component sets a corresponding atomic class on the div it renders. That atomic class applies a custom CSS property that maps to a design token\*.
```
Comment (chloerice) [inline documentation/components.md:11]:
```suggestion
When an atomic prop name is set on a component, the HTML renders with the corresponding atomic classes. For example, the marginTop prop accepts any of the spacing tokens. Setting the marginTop prop to 4 on a Box component sets a corresponding atomic class on the div it renders. That atomic class applies a custom CSS property that maps to a design token\*.
```
